libsoftwarecontainer/component-test/devicenodegateway_componenttest.cpp

```diff
@@ -97,3 +97,60 @@ TEST_F(DeviceNodeGatewayTest, TestOverwriteDeviceFails) {
     checkMode.wait();
     ASSERT_TRUE(checkMode.isSuccess());
 }
+
+struct testSetup
+{
+    const std::string firstConfiguration;
+    const std::string secondConfiguration;
+    const std::string deviceNameToExamine;
+    bool  shouldReconfigure;
+};
+
+class DynamicConfigurationTests :
+        public SoftwareContainerGatewayTest,
+        public  ::testing::WithParamInterface<testSetup>
+{
+public:
+    std::unique_ptr<DeviceNodeGateway> gw;
+
+    void SetUp() override
+    {
+        SoftwareContainerGatewayTest::SetUp();
+        gw = std::unique_ptr<DeviceNodeGateway>(new DeviceNodeGateway(m_container));
+    }
+};
+
+/*
+ * This data is fed to the DynamicConfiguration tests
+ */
+INSTANTIATE_TEST_CASE_P(DeviceModeConfigurations, DynamicConfigurationTests, ::testing::Values(
+        testSetup{
+            "[{\"name\": \"/dev/tty0\", \"mode\":  622}]",
+            "[{\"name\": \"/dev/tty0\", \"mode\":  755}]",
+            "/dev/tty0",
+            false},
```

Comment:
For readability, could you add a space before the opening braces and break the line for the closing braces and fix the indentation so the config strings are indented one level and the braces matches on the outer level?

Like:

testSetup {
    "string",
    "string"
},
testSetup {

etc.